File: mod-sources/com.s8.io.swgl/com/s8/io/swgl/scene/pipes/mat01/Mat01Appearance.java
package com.s8.io.swgl.scene.pipes.mat01;

import com.s8.api.objects.web.WebS8Session;
import com.s8.io.swgl.SWGL_Root;
import com.s8.io.swgl.material.SWGL_Material;
import com.s8.io.swgl.scene.pipes.SWGL_Appearance;


/**
 * 
 * @author pierreconvert
 *
 */
public class Mat01Appearance extends SWGL_Appearance {

	
	/**
	 * 
	 * @param branch
	 * @param typeName
	 */
	public Mat01Appearance(WebS8Session branch) {
		super(branch, SWGL_Root.WEB+"scene/pipes/mat01/Mat01Appearance");
	}
	
	
	
	/**
	 * 
	 * @param material
	 */
	public void setMaterial(SWGL_Material material) {
		vertex.fields().setObjectField("material", material);
	}
	

}

File: web-sources/s8-io-swgl/scene/pipes/mat01/Mat01Appearance.js
import { SWGL_Material } from "/s8-io-swgl/materials/SWGL_Material.js";
import { SWGL_Appearance } from "../SWGL_Appearance.js";


/**
 * 
 */
export class Mat01Appearance extends SWGL_Appearance {


    /**
     * @type {SWGL_Material}
     */
    material;


    constructor(){
        super();
    }



    /**
     * 
     * @param {SWGL_Material} material 
     */
    S8_set_material(material) {
        this.material = material;
    }


    S8_render(){
    }


    S8_dispose(){
    }

}
